Makefile.include

```diff
@@ -4,23 +4,43 @@
 all:
 
 # set undefined variables
-RIOTBASE ?= $(shell dirname "$(lastword $(MAKEFILE_LIST))")
-RIOTBASE := $(abspath $(RIOTBASE))
-
-CCACHE_BASEDIR := $(RIOTBASE)
-
-RIOTCPU ?= $(RIOTBASE)/cpu
-RIOTCPU := $(abspath $(RIOTCPU))
-
-RIOTBOARD ?= $(RIOTBASE)/boards
-RIOTBOARD := $(abspath $(RIOTBOARD))
-
-RIOTPKG ?= $(RIOTBASE)/pkg
+RIOTBASE       ?= $(shell dirname "$(lastword $(MAKEFILE_LIST))")
+CCACHE_BASEDIR ?= $(RIOTBASE)
+RIOTCPU        ?= $(RIOTBASE)/cpu
+RIOTBOARD      ?= $(RIOTBASE)/boards
+RIOTPKG        ?= $(RIOTBASE)/pkg
+RIOTPROJECT    ?= $(shell git rev-parse --show-toplevel 2>/dev/null || pwd)
```

Comment:
I may confuse something, but isn't `RIOTPROJECT` the same as `RIOTBASE`? Is there a case where `RIOTBASE` does not point to the RIOT root?